.github/copilot-instructions.md

```diff
@@ -0,0 +1,357 @@
+
+```bash
+# Start database only
+docker compose up -d db
+
+# Frontend development server (in separate terminal)
+cd front
+pnpm dev
+# Runs on http://localhost:3000
+
+# Backend development server (in separate terminal)
+cd back
+source ../.venv/bin/activate
+MYSQL_HOST=127.0.0.1 MYSQL_USER=root MYSQL_PASSWORD=dropapp_root MYSQL_DB=dropapp_dev MYSQL_PORT=32000 python -m boxtribute_server.dev_main
+# Runs on http://localhost:5005
+```
+
+## Testing Commands
+
+### Frontend Tests
+```bash
+# Run all frontend tests with coverage
+# NEVER CANCEL: Takes ~1-2 minutes to complete
+pnpm test:coverage
+# Timeout: Set to 5+ minutes
+
+# Run tests in watch mode for development
+pnpm test
+
+# Test shared components separately
+pnpm -C shared-components test:coverage
+```
+
+### Backend Tests
+```bash
+# Start database for testing
+docker compose up -d db
+
+# Install backend dependencies (REQUIRES internet access to PyPI)
+cd back
+source ../.venv/bin/activate
+pip install -U -e . -r requirements-dev.txt
+pre-commit install --overwrite
+# NOTE: May fail with network timeouts in restricted environments
+
+# Run backend tests (requires MySQL database and dependencies)
+pytest --ignore=test/auth0_integration_tests/
+# NEVER CANCEL: Takes ~5-10 minutes including database setup
+# Timeout: Set to 15+ minutes
+
+# Run auth0_integration_tests separately if needed (requires repository secrets)
+pytest test/auth0_integration_tests/
+# NOTE: Requires TEST_AUTH0_MANAGEMENT_API_CLIENT_SECRET from repository secrets
+
+# Run specific test categories (when dependencies are available)
+pytest test/unit_tests/          # Unit tests only (~2 minutes)
+pytest test/model_tests/         # Data model tests (~3 minutes)  
+pytest test/endpoint_tests/      # API endpoint tests (~8 minutes)
+```
+
+**CRITICAL NOTE**: Backend development requires internet access to PyPI for dependency installation. If you encounter `ReadTimeoutError` from pypi.org, document this in your PR and focus on frontend validation.
+
+## Build Commands
+
+### Frontend Build
+```bash
+# Build the frontend for production
+cd front
+pnpm build
+# Takes ~30-60 seconds
+# Timeout: Set to 3+ minutes
+
+# Build statistics app
+pnpm -C statviz build
+# Takes ~30-60 seconds
+```
+
+### Type Checking
+```bash
+# Check TypeScript types across all projects
+pnpm check-types
+# Takes ~40-60 seconds
+# Timeout: Set to 3+ minutes
+```
+
+## Code Quality Commands
+
+### Linting
+```bash
+# Lint all frontend code
+pnpm lint:all
+# Takes ~30-45 seconds
+
+# Lint and auto-fix
+pnpm lint:all:fix
+
+# Backend linting (via pre-commit)
+cd back
+source ../.venv/bin/activate
+pre-commit run --all-files
+# Takes ~30-60 seconds
+# NOTE: May fail with PyPI network timeouts in restricted environments
+```
+
+### Formatting
+```bash
+# Check code formatting
+pnpm format:check:all
+
+# Fix code formatting
+pnpm format:write:all
+# Takes ~3-5 seconds
+
+# Backend formatting is handled by pre-commit hooks (black, isort)
```

Comment:
actually you should run `git ls-files -- back | SKIP=generate-graphql-ts-types xargs pre-commit run --files`